src/utilities/TypeInfo.ts

```diff
@@ -190,15 +217,50 @@ export class TypeInfo {
       case Kind.ARGUMENT: {
         let argDef;
         let argType: unknown;
-        const fieldOrDirective = this.getDirective() ?? this.getFieldDef();
-        if (fieldOrDirective) {
-          argDef = fieldOrDirective.args.find(
-            (arg) => arg.name === node.name.value,
+        const directive = this.getDirective();
+        const fragmentSpread = this._fragmentSpread;
+        const fieldDef = this.getFieldDef();
+        if (directive) {
+          argDef = directive.args.find((arg) => arg.name === node.name.value);
+        } else if (fragmentSpread) {
+          const fragmentDef =
+            this._fragmentDefinitions[fragmentSpread.name.value];
+          const fragVarDef = fragmentDef?.variableDefinitions?.find(
+            (varDef) => varDef.variable.name.value === node.name.value,
           );
-          if (argDef) {
-            argType = argDef.type;
+          if (fragVarDef) {
+            const fragVarType = typeFromAST(schema, fragVarDef.type);
+            if (isInputType(fragVarType)) {
+              const fragVarDefault = fragVarDef.defaultValue
+                ? valueFromAST(fragVarDef.defaultValue, fragVarType)
+                : undefined;
+
+              // Minor hack: transform the FragmentArgDef
+              // into a schema Argument definition, to
+              // enable visiting identically to field/directive args
```

Comment:
Agreed.

It might be a bit more refactoring in the other methods that produce the expected type, but I'm curious if we can follow your suggestion where a "Signature" is just the union of ArgumentDef and VariableDef? That way we can avoid the risk of some subtle bug introduced by a virtual AST element here and keep the type of `argDef` as is, while still providing the correct `argType`?

In other words, the logic below adding to the "Default value" and "input type" stack is correct, but what needs to change to set _argument to undefined or to add a generalized "signature" getter?